operator/iamkeycontroller/delete.go

```diff
@@ -33,15 +39,36 @@ func (p *IAMKeyPipeline) deleteIAMKey(ctx *pipelineContext) error {
 	log := controllerruntime.LoggerFrom(ctx)
 	iamKey := ctx.iamKey
 
-	err := p.exoscaleClient.RevokeIAMAccessKey(ctx, iamKey.Spec.ForProvider.Zone, &exoscalesdk.IAMAccessKey{
-		Key: &iamKey.Status.AtProvider.KeyID,
-	})
-	if err != nil {
-		return err
+	log.Info("Starting IAM key deletion", "keyName", iamKey.Spec.ForProvider.KeyName)
+
+	fmt.Println(iamKey.ObjectMeta.Annotations)
+
+	if _, ok := iamKey.ObjectMeta.Annotations[RoleIDAnnotationKey]; ok {
+
+		_, err := ExecuteRequest(ctx, "DELETE", ctx.iamKey.Spec.ForProvider.Zone, "/v2/api-key/"+iamKey.Status.AtProvider.KeyID, p.apiKey, p.apiSecret, nil)
```

Comment:
yeah, that happened because:
1) exoscale creates a role and returns us ID
2) You can immediately GET this role by ID, but...
3) You couldn't create resource referencing to this role-id immediately (they have probably some sync on datacenters)

I added small for loop which executes query and blocks for 500ms after fail as I find it cleaner solution than regular time.Sleep(time.Second * 10)

Manual tests shows issue is solved

Reply:
That should be fixed right now